elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java

```diff
@@ -351,11 +352,16 @@ private Map<String, Object> getMappingChanges(final Map<String, Object> template
                                                                  key, value.rightValue(), value.leftValue()));
 
             // Check if 'type' has changed for object/nested properties
-            final Map<String, ValueDifference<Object>> typeDifferences = entriesDiffering.entrySet().stream()
-                .filter(e -> ((Map<?, ?>) (e.getValue().leftValue())).containsKey(MAPPING_PROPS_KEY)
-                && (((Map<?, ?>) (e.getValue().leftValue())).containsKey(MAPPING_TYPE_KEY)
-                || ((Map<?, ?>) (e.getValue().rightValue())).containsKey(MAPPING_TYPE_KEY)))
-                .collect(Collectors.toMap(e -> e.getKey(), e -> e.getValue()));
+            final Map<String, ValueDifference<Object>> typeDifferences = new LinkedHashMap<>();
+          
+            for (final Map.Entry<String, ValueDifference<Object>> e : entriesDiffering.entrySet()) {
+                final Map<?, ?> leftEntry = ((Map<?, ?>) (e.getValue().leftValue()));
+                final Map<?, ?> rightEntry = ((Map<?, ?>) (e.getValue().rightValue()));
+                if (leftEntry.containsKey(MAPPING_PROPS_KEY)
+                    && (!leftEntry.get(MAPPING_TYPE_KEY).equals("object") || !rightEntry.get(MAPPING_TYPE_KEY).equals("object"))) {
+                    typeDifferences.put(e.getKey(), e.getValue());
+                }
```

Comment:
This is the same operation. 
Previously the `type` field wasn't included in the mapping if it was an `object` as this is the default.
Now, when serializing via `getObjectAsHashMap` method, the `type` field is included and set to `object`, i.e.
**new mapping**
{type=object, properties={CODE={type=keyword, ignore_above=10922,}, CONFIDENCE={type=double, store=true}}}
**old mapping**
{properties={CODE={type=keyword, ignore_above=10922,}, CONFIDENCE={type=double, store=true}}}